2016/proposals/chris-medrela.md

```diff
@@ -0,0 +1,461 @@
+# Manage workflow for Software/Data Carpentry instructor training
+
+## Abstract
+
+Software Carpentry (SWC) has a workflow for Instructor Trainings. Currently, it uses Etherpads and Google Docs spreadsheets, which slows down SWC Admins and Trainers whose time is already precious. The goal is to extend AMY (Django-base web application for SWC Admins) to manage the process of training people to be instructors, to avoid manual data copying and synchronization and make the entire process smoother.
+
+[The appendix](#appendix) at the end of this proposal contains [description of the current workflow] and [list of issues] based on [Greg Wilson's pull request] and its discussion. Also, whenever something is written Title-cased, you can look it up in the [glossary](#glossary) in the [appendix](#appendix) at the end of this proposal. Hopefully, the glossary will keep this proposal unambiguous. It also serves as a list of all [user roles] and [resources](#resources). The latter is also based on the Greg's pull request.
+
+[Greg Wilson's pull request]: https://github.com/swcarpentry/board/pull/81/files
+[user roles]: people-or-user-roles
+
+Here is [#89](https://github.com/numfocus/gsoc/issues/89) issue where I discussed this proposal. 
+
+I split this proposal into as small milestones as possible. Every milestone ends up with new version of Amy being pushed to production and introduced to all its users. This minimizes the chances of failure and let us adapt or change almost everything after every milestone -- even if it means changing the scope of this project. The milestones are:
+
+1. [Milestone 1: Managing SWC Checkout Process](#milestone-1-managing-swc-checkout-process) -- that is Homework, Discussion Session, Checkout Session and welcoming to the community.
+
+1. [Milestone 2: Managing DC Checkout Process](#milestone-2-managing-dc-checkout-process)
+
+1. [Milestone 3: Managing Recruiting Candidates](#milestone-3-managing-recruiting-candidates) and rest of the workflow, that is matching them to Instructor Trainings and running those trainings.
+
+It may be necessary to move from SQLite to Postgress. I'm discussing in [#716](https://github.com/swcarpentry/amy/issues/716) with Piotr Banaszkiewicz if it's really necessary, but if it is, it must be done at the beginning, as an additional milestone.
+
+I believe it's really important to get feedback as soon as possible, therefore:
+
+- Before introducing new features to all users, they'll be tested by a group of volunteering early users. That way, we don't have to test everything before showing it to first users, so we can get feedback faster and iterate quicker.
```

Comment:
In that case, the only thing we can do is to extensively test the code for different user scenarios before exposing new functionality to all users.